source/module_base/test/math_ylmreal_test.cpp

```diff
@@ -86,7 +195,10 @@ class YlmRealTest : public testing::Test
         g[3].set(-1.0,-1.0,-1.0);
 
         rly = new double[nylm];
-        ref = new double[nylm*ng]{
+        rlyvector.resize(nylm);
+        rlgy = new double[nylm][3];
+        rlgyvector.resize(nylm,std::vector<double>(3));
+        ref = new double[64*4]{
```

Comment:
It seems that all variables in `SetUp`  is of constant value.

Reply:
yes

Reply:
> yes

If the allocated arrays are not been used in the following part for pointer usage (e.g. realloc), it is better to put them in class.

Reply:
https://stackoverflow.com/questions/27408819/why-cant-non-static-data-members-be-constexpr
@pxlxingliang FYI